src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.BufferToTagTree.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Tagging;
+
+internal partial class AbstractAsynchronousTaggerProvider<TTag>
+{
+    private readonly struct BufferToTagTree(ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>> map)
```

Comment:
i create this wrapper so we could have a simpler alias than writing out `ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>>` everywhere.  I couldn't use real aliases due to the use of hte TTag type parameter.  

I don't love this *especially* because of the InterlockedExchange helpers below.

Reply:
I can't say I love this either.

Reply:
ok.  im' going to remove.

Reply:
removed.